tests/src/FunctionalJavascript/ContributionPageTest.php

```diff
@@ -0,0 +1,140 @@
+    $element_form->pressButton('Save');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+
+    // Put contact elements into new page.
+    $contact_information_page_row_handle = $this->assertSession()->elementExists('css', '[data-drupal-selector="edit-webform-ui-elements-contact-information"] a.tabledrag-handle');
+    $contact_information_page_row_handle->keyDown(38);
+    $contact_information_page_row_handle->keyUp(38);
+    $contact_information_page_row_handle->keyDown(38);
+    $contact_information_page_row_handle->keyUp(38);
+    $contact_information_page_row_handle->blur();
+    $contact_fieldset_row_handle = $this->assertSession()->elementExists('css', '[data-drupal-selector="edit-webform-ui-elements-civicrm-1-contact-1-fieldset-fieldset"] a.tabledrag-handle');
+    $contact_fieldset_row_handle->keyDown(39);
+    $contact_fieldset_row_handle->keyUp(39);
+    $contact_fieldset_row_handle->blur();
+    $this->getSession()->getPage()->pressButton('Save elements');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+
+    $this->drupalGet($this->webform->toUrl('canonical'));
+    $error_messages = $this->getSession()->getPage()->findAll('css', '.messages.messages--error');
+    $this->assertCount(0, $error_messages, implode(', ', array_map(static function(NodeElement $el) {
+      return $el->getValue();
+    }, $error_messages)));
+    $this->htmlOutput();
+    $this->getSession()->getPage()->fillField('First Name', 'Frederick');
+    $this->getSession()->getPage()->fillField('Last Name', 'Pabst');
+    $this->getSession()->getPage()->fillField('Email', 'fred@example.com');
+    $this->getSession()->getPage()->pressButton('Next >');
+    $this->htmlOutput();
+    $this->getSession()->getPage()->fillField('Contribution Amount', '25.00');
+    $this->assertSession()->elementExists('css', '#wf-crm-billing-items');
+    $this->assertSession()->elementTextContains('css', '#wf-crm-billing-total', '25.00');
+
+    // Wait for the credit card form to load in.
+    $this->assertSession()->waitForField('credit_card_number');
+    $this->getSession()->getPage()->fillField('Card Number', '4111111111111111');
+    $this->getSession()->getPage()->fillField('Security Code', '123');
+    $this->getSession()->getPage()->selectFieldOption('credit_card_exp_date[M]', '11');
+    $this->getSession()->getPage()->selectFieldOption('credit_card_exp_date[Y]', '2023');
+    $this->getSession()->getPage()->fillField('Billing First Name', 'Frederick');
+    $this->getSession()->getPage()->fillField('Billing Last Name', 'Pabst');
+    $this->getSession()->getPage()->fillField('Street Address', '123 Milwaukee Ave');
+    $this->getSession()->getPage()->fillField('City', 'Milwaukee');
+    $this->getSession()->getPage()->fillField('Country', '1228');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    // @todo find a better way to wait for select2.
+    sleep(1);
+    $this->getSession()->getPage()->fillField('State/Province', '1048');
```

Comment:
This doesn't work because Select2 hides the actual element. This needs to be changed to interact with Select2 properly.